HealthcareApp/src/MigrationAdminPanel/Services/JsonMigrationsService.cs

```diff
@@ -0,0 +1,39 @@
+using Application.Abstractions.Decorators;
+using Application.Abstractions;
+using Domain.Entities;
+using MigrationAdminPanel.Abstractions;
+using MigrationAdminPanel.DataTypes;
+using Newtonsoft.Json;
+
+namespace MigrationAdminPanel.Services
+{
+    public class JsonMigrationsService : MigrationsService
+    {
+        private readonly IAppointmentRepository _appointmentRepository;
+
+        public JsonMigrationsService(IUserManagerDecorator<ApplicationUser> userManager, IAppointmentRepository appointmentRepository)
+            : base(userManager)
+        {
+            _appointmentRepository = appointmentRepository;
+        }
+
+        protected override async Task<object?> ReadDataFromFile(string path)
+        {
+            string jsonString = await File.ReadAllTextAsync(path);
+            return JsonConvert.DeserializeObject<JsonMigrationsData>(jsonString);
+        }
```

Comment:
using)

Reply:
How here I could apply using, I use here only static methods

Reply:
Vlad, it's better to use StreamReader because if the files are large, there will be issues. However, if the files are small, your code looks fine

Reply:
I think it's about reading the file in stream. We can omit it for this task since the task is more about working with DB. Thanks Nazar